ui/dashboards/src/components/GridLayout/Row.tsx

```
@@ -0,0 +1,154 @@
+              ? undefined
+              : { isOpen, onToggleOpen: () => setIsOpen((current) => !current) }
+          }
+        />
+      )}
+      <Collapse in={isOpen} unmountOnExit appear={false} data-testid="panel-group-content">
+        <ResponsiveGridLayout
+          className="layout"
+          breakpoints={{ [GRID_LAYOUT_SMALL_BREAKPOINT]: theme.breakpoints.values.sm, xxs: 0 }}
+          cols={GRID_LAYOUT_COLS}
+          rowHeight={ROW_HEIGHT}
+          draggableHandle=".drag-handle"
+          resizeHandles={['se']}
+          isDraggable={isEditMode && !hasViewPanel}
+          isResizable={isEditMode && !hasViewPanel}
+          margin={[DEFAULT_MARGIN, DEFAULT_MARGIN]}
+          containerPadding={[0, 10]}
+          layouts={{ sm: itemLayouts }}
+          onLayoutChange={onLayoutChange}
+          onWidthChange={onWidthChange}
+          allowOverlap={hasViewPanel} // Enabling overlap when viewing a specific panel because panel in front of the viewed panel will add empty spaces (empty row height)
+        >
+          {itemLayouts.map(({ i, w }) => (
+            <div
+              key={i}
+              style={{
+                display: itemLayoutViewed !== undefined ? (itemLayoutViewed === i ? 'unset' : 'none') : 'unset',
+              }}
+            >
+              <ErrorBoundary FallbackComponent={ErrorAlert}>
+                <GridItemContent
+                  panelOptions={panelOptions}
+                  panelGroupItemId={{ panelGroupId, panelGroupItemLayoutId: i, repeatVariable }}
+                  width={calculateGridItemWidth(w, gridColWidth)}
+                />
+              </ErrorBoundary>
+            </div>
+          ))}
+        </ResponsiveGridLayout>
+      </Collapse>
+    </GridContainer>
+  );
+}
+
+const calculateGridItemWidth = (w: number, colWidth: number): number => {
+  // 0 * Infinity === NaN, which causes problems with resize contraints
+  if (!Number.isFinite(w)) return w;
```

I think such a case should never happen, and if it happens, an exception should be thrown. 
Considering the usage of the `calculateGridItemWidth`, it is likely Infinity or -Infinity is passed to width here
```typescript
  <GridItemContent
                  panelOptions={panelOptions}
                  panelGroupItemId={{ panelGroupId, panelGroupItemLayoutId: i, repeatVariable }}
                  width={calculateGridItemWidth(w, gridColWidth)}
                />
```

We should probably throw an exception here

```typescript
if (!Number.isFinite(w) || w < 0) {
    throw new Error("Invalid value for 'w'. It must be a finite, non-negative number.");
  }
```